app/src/main/java/com/android/doctorapp/ui/doctordashboard/doctorfragment/DoctorAddressFragment.kt

```diff
@@ -0,0 +1,256 @@
+                            } else {
+                                Log.d("TAG", "onCreateView: else")
+                            }
+
+                        }
+                    }
+
+                }
+        }
+
+        return binding {
+            viewModel = this@DoctorAddressFragment.viewModel
+            lifecycleOwner = viewLifecycleOwner
+        }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setUpWithViewModel(viewModel)
+    }
+
+    fun requestLocationUpdates() {
+        if (ActivityCompat.checkSelfPermission(
+                requireActivity(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                requireActivity(),
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            return
+        }
+
+        if (context?.isGPSEnabled() == true) {
+            fusedLocationClient.requestLocationUpdates(
+                locationRequest,
+                locationCallback,
+                null
+            )
+        } else {
+            GpsUtils(requireContext()).turnGPSOn(object : GpsUtils.onGpsListener {
+                override fun gpsStatus(isGPSEnable: Boolean) {
+                }
+            })
+        }
+    }
+
+    private val locationCallback = object : LocationCallback() {
+        override fun onLocationResult(locationResult: LocationResult?) {
+            locationResult?.lastLocation?.let { location ->
+                // Handle the location update here
+                latitude = location.latitude
+                longitude = location.longitude
+
+                val addresses: List<Address>?
+                val geocoder = Geocoder(requireContext(), Locale.getDefault())
+                addresses = geocoder.getFromLocation(latitude, longitude, 1)
+                val address: String = addresses!![0].getAddressLine(0)
+                val hash = GeoFireUtils.getGeoHashForLocation(GeoLocation(latitude, longitude))
+
+                val updates: MutableMap<String, Any> = mutableMapOf(
+                    "geohash" to hash,
+                    "latitude" to latitude,
+                    "longitude" to longitude,
+                )
+                viewModel.addressLatLngList.value = updates
+//                Log.d("TAG", "onLocationResult 156 : $updates")
+
+
+                findNavController().previousBackStackEntry?.savedStateHandle?.set("address", address)
+                findNavController().previousBackStackEntry?.savedStateHandle?.set("addressLatLng", updates)
+                fusedLocationClient.removeLocationUpdates(this)
+                findNavController().popBackStack()
+
+                // Find cities within 50km of London
```

Comment:
Please remove all commented code and check all things that need to be reduce code wise.

Reply:
Done